java-cfenv/build.gradle

```diff
@@ -12,15 +12,15 @@ dependencies {
 
     testImplementation "org.springframework.boot:spring-boot-starter-test"
     testImplementation "junit:junit"
-    testImplementation "org.jmockit:jmockit:${jmockitVersion}"
+    testImplementation "com.github.hazendaz.jmockit:jmockit:${jmockitVersion}"
 
     testRuntimeOnly('org.junit.vintage:junit-vintage-engine') {
         exclude group: 'org.hamcrest', module: 'hamcrest-core'
     }
 
     testFixturesApi platform("org.springframework.boot:spring-boot-dependencies:${springBootVersion}")
     testFixturesImplementation "org.springframework.boot:spring-boot-starter-test"
-    testFixturesImplementation "org.jmockit:jmockit:${jmockitVersion}"
+    testFixturesImplementation "com.github.hazendaz.jmockit:jmockit:${jmockitVersion}"
```

Comment:
This seems a bit obscure repor. Isn't it possible to replace by mockito? The latter is well-known and publicly maintained.

Reply:
yeah, well, jmockit does not support Java 17 in combination with Jacoco...
https://github.com/jmockit/jmockit1/issues/615

Indeed, going mockito is the long term solution - let me see if I can make it work in this PR

Reply:
so I put 2hours into it and .... it's complicated...
jmockit is brought into the mix to override `System.getEnv()` which is only possible via reflection.

Most of the tests simulate the presence of VCAP_SERVICES and VCAP_APPLICATION via the class CFEnvMock

I'm gonna try and change this class and add allow the JVM 17 to use reflection for tests.

To be continued!

OR

we put that in another PR to unblock this PR